From: Kentaro Hayashi <kenhys@gmail.com>
Date: Wed, 16 Oct 2024 20:05:24 +0900
Subject: Change compiler from clang to gcc

Description: Change compiler from clang to gcc
  Debian default c++ compiler is g++. This changes to it.
Forwarded: not-needed
Last-Update: 2022-05-05

It seems that upstream fixes it in another approach.
https://github.com/google/mozc/commit/aba090da42a8366e72bc320dfafc3f9f93755edf


Signed-off-by: Nobuhiro Iwamatsu <iwamatsu@debian.org>
Signed-off-by: Kentaro Hayashi <kenhys@gmail.com>
---
 src/gyp/common.gypi | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/src/gyp/common.gypi b/src/gyp/common.gypi
index 39b0ce0..e9756d2 100644
--- a/src/gyp/common.gypi
+++ b/src/gyp/common.gypi
@@ -105,7 +105,7 @@
     ],
     # Libraries for GNU/Linux environment.
     'linux_ldflags': [
-      '-lc++',
+      '-lstdc++',
       '-pthread',
     ],
 
@@ -117,10 +117,10 @@
         'compiler_host_version_int': 303,  # Clang 3.3 or higher
       }],
       ['target_platform=="Linux"', {
-        'compiler_target': 'clang',
-        'compiler_target_version_int': 304,  # Clang 3.4 or higher
-        'compiler_host': 'clang',
-        'compiler_host_version_int': 304,  # Clang 3.4 or higher
+        'compiler_target': 'g++',
+        'compiler_target_version_int': 409,  # gcc 4.9 or higher
+        'compiler_host': 'g++',
+        'compiler_host_version_int': 304,  # gcc 4.9 or higher
       }],
     ],
   },
@@ -339,8 +339,8 @@
     ['target_platform=="Linux"', {
       'make_global_settings': [
         ['AR', '<!(which ar)'],
-        ['CC', '<!(which clang)'],
-        ['CXX', '<!(which clang++)'],
+        ['CC', '<!(command -v gcc)'],
+        ['CXX', '<!(command -v g++)'],
         ['LD', '<!(which ld)'],
         ['NM', '<!(which nm)'],
         ['READELF', '<!(which readelf)'],
